io/runes/runes_test.go

```diff
@@ -0,0 +1,177 @@
+package runes_test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+	"testing"
+
+	"github.com/madlambda/spells/assert"
+	"github.com/madlambda/spells/io/runes"
+	"github.com/madlambda/spells/iotest"
+)
+
+type testResult struct {
+	err     error
+	partial []rune
+}
+
+type testcase struct {
+	name   string
+	input  string
+	repeat int
+	want   testResult
+}
+
+const socraticParadox = "Εν οίδα ότι ουδέν οίδα"
+
+var testcases = []testcase{
```

Comment:
sorry for duplicating the testcases for utf8 and runes. Errors are treated differently and for utf8 I wanted to also test the undecided bytes returned in the error.